cmd/node_problem_detector.go

```diff
@@ -46,23 +47,27 @@ func main() {
 
 	npdo.ValidOrDie()
 
-	monitors := make(map[string]types.Monitor)
-	for _, config := range npdo.SystemLogMonitorConfigPaths {
-		if _, ok := monitors[config]; ok {
-			// Skip the config if it's duplicated.
-			glog.Warningf("Duplicated monitor configuration %q", config)
-			continue
-		}
-		monitors[config] = systemlogmonitor.NewLogMonitorOrDie(config)
+	// Initialize problem daemons.
+	problemDaemons, errors := problemdaemon.NewProblemDaemons(npdo.MonitorsConfigs)
+	for _, err := range errors {
+		glog.Errorf("Problem daemon initialization failed: %v", err)
+	}
+	if len(errors) != 0 && npdo.PanicOnProblemDaemonFailure {
+		glog.Fatalf("Quit NPD because of above problem daemon initialization failures")
 	}
 
+	// Legacy per-monitor configuration.
+	// This section is kept so that existing command line option still works on NPD.
+	// TODO(xueweiz): when NPD's refactoring is finished, remove legacy configuration.
+	for _, config := range npdo.SystemLogMonitorConfigPaths {
+		problemDaemons = append(problemDaemons, systemlogmonitor.NewLogMonitorOrDie(config))
+	}
 	for _, config := range npdo.CustomPluginMonitorConfigPaths {
```

Comment:
ditto.

Reply:
Will do if we decide to merge the argument holders.